### PR TITLE
Replace unused raise argument with a dummy value

### PR DESCRIPTION
**Pull request by chambart**
When compiling with `-g`
```OCaml
   try assert (0 = 40 + 2) with _ -> ()
```
the exception handler is kept, but the exception symbol is marked as unused by `Data_flow` because the exception value is actually unused. But we can't remove the argument because exceptions arguments can't be removed.

### CODE REVIEW

File: middle_end/flambda2/simplify/simplify_let_cont_expr.ml
Comment (Gbury) [inline middle_end/flambda2/simplify/simplify_let_cont_expr.ml:121]:
```suggestion
                 the moment represent this transformation, so
```

File: middle_end/flambda2/simplify/simplify_apply_cont_expr.ml
Comment (Gbury) [inline middle_end/flambda2/simplify/simplify_apply_cont_expr.ml:86]: Maybe we this should be named something like `exn_arg` or `exn_bucket` to explicit the fact that it is the exception being raised ?

**Review by Gbury — CHANGES_REQUESTED** [review]
There is a typo (see comment), but otherwise this looks good.
Is possible, could you add your example as a regression test in the testsuite ?

**Comment by mshinwell** [comment]
Please don't merge this until after I've merged a forthcoming patch that significantly rearranges some of the code in the simplifier, in preparation for ocamlformat.

**Review by Gbury — APPROVED** [review]

**Review by lthls — APPROVED** [review]

**Pull request merged**
